src/toil/utils/toilStatus.py

```diff
@@ -64,96 +89,157 @@ def main():
+                # Traverse jobs in stack
+                for jobs in jobGraph.stack:
+                    for successorJobStoreID in [x.jobStoreID for x in jobs]:
+                        if (successorJobStoreID not in foundJobStoreIDs and jobStore.exists(successorJobStoreID)):
+                            inner(jobStore.load(successorJobStoreID))
+    
+                # Traverse service jobs
+                for jobs in jobGraph.services:
+                    for serviceJobStoreID in [x.jobStoreID for x in jobs]:
+                        if jobStore.exists(serviceJobStoreID):
+                            assert serviceJobStoreID not in foundJobStoreIDs
+                            foundJobStoreIDs.add(serviceJobStoreID)
+                            jobsToReport.append(jobStore.load(serviceJobStoreID))
+            inner(jobGraph)
+    
+        logger.info('Traversing the job graph to find the jobs. This may take a couple minutes.')
+        traverseGraph(rootJob)
 
-    failedJobs = []
+    else:
+        for jobID in options.jobs:
+            try:
+                jobsToReport.append(jobStore.load(jobID))
+            except JobException:
+                print('The job %s could not be found.' % jobID,
+                      file=sys.stderr)
+                sys.exit(0)
+    
+    ##########################################
+    # Report on the jobs
+    ########################################## 
+    
     hasChildren = []
+    readyToRun = []
+    zombies = []
+    hasLogFile = []
     hasServices = []
     services = []
-    currentlyRunnning = []
-
-    for job in totalJobs:
+    
+    
+    for job in jobsToReport:
+        properties = set()
         if job.logJobStoreFileID is not None:
-            failedJobs.append(job)
-        if job.stack:
+            hasLogFile.append(job)
+        
+        childNumber =  reduce(lambda x, y : x + y, map(len, job.stack) + [0])
+        if childNumber > 0: # Total number of successors > 0
             hasChildren.append(job)
-        elif job.remainingRetryCount != 0 and job.logJobStoreFileID != 0 and job.command:
-            # The job has no children, hasn't failed, and has a command to run. This indicates that the job is
-            # likely currently running, or at least could be run.
-            currentlyRunnning.append(job)
+            properties.add("HAS_CHILDREN")
+            
+        elif job.command != None:
+            #if job.remainingRetryCount == 0 and job.logJobStoreFileID != None 
+            # The job has no children and has a command to run. This indicates that the job could be run.
+            readyToRun.append(job)
+            properties.add("READY_TO_RUN")
+        
+        else:
+            # The job has no successors and no command, so is a zombie job
+            zombies.append(job)
+            properties.add("IS_ZOMBIE")
+            
         if job.services:
             hasServices.append(job)
+            properties.append("HAS_SERVICES")
```

Comment:
Use `.add` here to avoid a crash (`.append` isn't defined for a set)

Reply:
Fixed.  Really good to know.  I wish PyCharm had thrown an error.